src/java/org/apache/cassandra/cql3/SingleColumnRelation.java

```diff
@@ -38,127 +37,105 @@
 import static org.apache.cassandra.cql3.statements.RequestValidations.invalidRequest;
 
 /**
- * Relations encapsulate the relationship between an entity of some kind, and
- * a value (term). For example, {@code <key> > "start" or "colname1" = "somevalue"}.
- *
+ * A filter on a column values.
  */
 public final class SingleColumnRelation extends Relation
 {
-    private final ColumnIdentifier entity;
+    private final ColumnIdentifier column;
     private final Term.Raw mapKey;
     private final Term.Raw value;
-    private final List<Term.Raw> inValues;
+    private final Terms.Raw inValues;
 
-    private SingleColumnRelation(ColumnIdentifier entity, Term.Raw mapKey, Operator type, Term.Raw value, List<Term.Raw> inValues)
+    private SingleColumnRelation(ColumnIdentifier column, Term.Raw mapKey, Operator operator, Term.Raw value, Terms.Raw inValues)
```

Comment:
Nit: I'd add `@Nullable` in the optional arguments, and maybe also in the class attributes:
```suggestion
    private SingleColumnRelation(ColumnIdentifier column,
                                 @Nullable Term.Raw mapKey,
                                 Operator operator,
                                 @Nullable Term.Raw value,
                                 @Nullable Terms.Raw inValues)
```